src/Operator/Constructors.jl

```diff
@@ -780,6 +879,184 @@ function advectionoperator(basis::AbstractBasis, mesh::Mesh, wind = [1, 1])
     advection = Operator(advectionweakform, mesh, inputs, outputs)
     return advection
 end
+
+"""
+```julia
+supgadvectionoperator(basis, mesh)
+```
+Convenience constructor for SUPG advection operator
+
+# Weak form: Right hand side
+- ``\\int wind \\nabla v u - wind wind τ \\nabla v \\nabla u``
+
+# Arguments:
+- `basis`: basis for all operator fields to use
+- `mesh`:  mesh for operator
+- `wind`:  advection speed in 2D
+- `τ`:     scaling for SUPG
+
+# Returns:
+- SUPG advection operator with basis on mesh
+
+# Example:
+```jldoctest
+# supg advection operator
+mesh = Mesh2D(1.0, 1.0);
+mapping = nothing
+basis = TensorH1LagrangeBasis(3, 4, 1, mesh.dimension, collocatedquadrature = false, mapping = mapping)
+wind = [1, 1]
+supgadvection = LFAToolkit.supgadvectionoperator(basis, mesh);
+
+# verify
+println(supgadvection)
+
+# output
+
+finite element operator:
+2d mesh:
+    dx: 1.0
+    dy: 1.0
+
+2 inputs:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation modes:
+    interpolation
+    gradient
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    quadratureweights
+
+1 output:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    gradient
+```
+"""
+P = 2
+τ = 0.5 / (P - 1) # Tau scaling for SUPG, 0 returns Galerkin method
+function supgadvectionoperator(basis::AbstractBasis, mesh::Mesh, wind = [1, 1])
```

Comment:
I think we need a way for a call to `GalleryOperator()` to set this.

Perhaps we aim for an interface like
```julia
GalleryOperator("supgadvection", 2, 3, mesh; parameters=(wind=[1, 2]))
```

with us updating line 182 to
```
return operatorgallery[name](basis, mesh, parameters)
```

and using something like this for `supgadvectionoperator`'s definition

```
function supgadvectionoperator(basis::AbstractBasis, mesh::Mesh, parameters=(wind=[1, 1]))
```

I'm not exactly sure how argument destructuring and kwargs mix and match with Julia, but that's the basic idea I'm thinking of for an interface

Reply:
alright, thanks for the clarification.